### ISSUE TITLE
Experssions in GROUP BY have wrong expression index

### ISSUE THREAD

**Issue by Enter-tainer** [singularity-data/risingwave#2467]
**Describe the bug**
As title, there are 2 duplicate `expr#0` in the following example.
```
dev=>  select 1, (a+1) from t group by a+1;
 expr#0 | expr#0 
--------+--------
      1 |      3
      1 |      2
(2 rows)
```

And for the following test case, the column of the group key has the wrong expression index either:

```
dev=>  select 1, 2, (a+1), 3 from t group by a+1;
 expr#0 | expr#1 | expr#0 | expr#3 
--------+--------+--------+--------
      1 |      2 |      3 |      3
      1 |      2 |      2 |      3
(2 rows)
```

**To Reproduce**
```
create table t1(a int);
select 1, (a+1) from t1 group by a+1;
```

**Expected behavior**
The group key column should be `expr#1` rather than `expr#0`.

**Actual Behavior**
```
dev=> create table t1(a int);
CREATE_TABLE
dev=>  select 1, (a+1) from t1 group by a+1;
 expr#0 | expr#0 
--------+--------
(0 rows)
```

**Comment by xiangjinwu** [comment]
Just adding some context:
After #1433 is resolved, these `expr#0` will be internal names for debugging only. It is still worth "fixing", but will not be a user visible thing.

The user visible names will be `?column?` - just like PostgreSQL.

**Comment by xiangjinwu** [comment]
```
dev=> explain select 2*3, (a+1) from t group by a+1;
                       QUERY PLAN                        
---------------------------------------------------------
 BatchExchange { order: [], dist: Single }
   BatchProject { exprs: [(2:Int32 * 3:Int32), $0] }         -- [Proj1Expr0, Proj0Expr0]
     BatchHashAgg { group_keys: [$0], aggs: [] }             -- [Proj0Expr0]
       BatchExchange { order: [], dist: HashShard([0]) }
         BatchProject { exprs: [($0 + 1:Int32)] }            -- [Proj0Expr0]
           BatchScan { table: t, columns: [a] }
(6 rows)
```

These 2 `expr#0` are generated by 2 different project nodes, and the higher one also reuses the name passed from lower one, resulting in 2 `expr#0` at the top. We could have prepended PlanNodeId to differentiate them just like diagram above.

At this point, we are not trying to have unique schema field names for **internal nodes**. As mentioned above, **user-visible names** (the ones shown in original bug report) have been decoupled and is the same as PostgreSQL.

In a **materialized view**, user visible names follow the same rule and aliasing is required (#2436), while **hidden column** names are also dedup'ed (#2662).

```
dev=> explain create materialized view mv1 as select 1 as z from t group by 2+3, 4+5 order by 6+7;
                                                                          QUERY PLAN                                                                           
---------------------------------------------------------------------------------------------------------------------------------------------------------------
 StreamMaterialize { columns: [z, expr#1(hidden), expr#0(hidden), expr#1#1(hidden)], pk_columns: [expr#0, expr#1#1], order_descs: [expr#1, expr#0, expr#1#1] }
   StreamProject { exprs: [1:Int32, (6:Int32 + 7:Int32), $0, $1] }
     StreamHashAgg { group_keys: [$0, $1], aggs: [count] }
       StreamExchange { dist: HashShard([0, 1]) }
         StreamProject { exprs: [(2:Int32 + 3:Int32), (4:Int32 + 5:Int32), $0] }
           StreamTableScan { table: t, columns: [_row_id], pk_indices: [0] }
(6 rows)
```
